src/closure.rs

```diff
@@ -53,23 +59,28 @@ impl Closure {
         }
 
         unsafe extern "C" fn finalize<F>(notify_data: *mut c_void, _closure: *mut gobject_ffi::GClosure)
-            where F: Fn(&[Value]) -> Option<Value> + Send + Sync + 'static
+            where F: Fn(&[Value]) -> Option<Value>
         {
             let _callback: Box<F> = Box::from_raw(notify_data as *mut _);
             // callback is dropped here.
         }
 
-        unsafe {
-            let size = 4 + 4 + 3 * mem::size_of::<*mut c_void>() as u32;
-            let closure = gobject_ffi::g_closure_new_simple(size, ptr::null_mut());
-            assert_ne!(closure, ptr::null_mut());
-            let callback = Box::new(callback);
-            let ptr: *mut F = Box::into_raw(callback);
-            let ptr: *mut c_void = ptr as *mut _;
-            gobject_ffi::g_closure_set_meta_marshal(closure, ptr, Some(marshal::<F>));
-            gobject_ffi::g_closure_add_finalize_notifier(closure, ptr, Some(finalize::<F>));
-            from_glib_none(closure)
-        }
+        // Due to bitfields we have to do our own calculations here for the size of the GClosure:
+        // - 4: 32 bits in guint bitfields at the beginning
+        // - padding due to alignment needed for the following pointer
+        // - 3 * size_of<*mut c_void>: 3 pointers
+        // We don't store any custom data ourselves in the GClosure
+        let size = 4
+            + (mem::align_of::<*mut c_void>() as u32 - 4)
+            + 3 * mem::size_of::<*mut c_void>() as u32;
```

Comment:
maybe better use max(4, mem::aligin_of) ?

Reply:
No, that would be wrong. Consider for example 16 byte alignment for pointers, we would have to add 12 bytes of padding here and not 16 (with 16 bytes of padding we would be unaligned again).

Or take 4 byte alignment: no padding would have to be added, but you would add 4 bytes then. Which does not cause problems in our specific case but is not needed either.

Reply:
I does mean only instead first 2 addendums,
but this also not error prone if "align_of<*void>" < 4 same as your case.
Maybe add assert about this?

Reply:
> I does mean only instead first 2 addendums,

I don't understand what you mean, can you write the whole calculation you think would be better?

We need 4 bytes for the bitfields, N bytes padding, then 3 pointers.

> but this also not error prone if "align_of<*void>" < 4 same as your case.
> Maybe add assert about this?

I'm not aware of any such platform but that's a good idea nonetheless.

Reply:
Ah indeed, that's slightly nicer and then we also don't need the assert. Thanks! Will update when I'm back at the computer.

Reply:
Done :)